scipy/stats/_continuous_distns.py

```diff
@@ -7040,11 +7040,25 @@ def _stats(self, df):
     def _entropy(self, df):
         if df == np.inf:
             return norm._entropy()
+
         half = df/2
         half1 = (df + 1)/2
-        return (half1*(sc.digamma(half1) - sc.digamma(half))
-                + np.log(np.sqrt(df)*sc.beta(half, 0.5)))
 
+        # _lazywhere complains if the functions don't have inputs
+        def regular(df):
+            return (half1*(sc.digamma(half1) - sc.digamma(half))
+                    + np.log(np.sqrt(df)*sc.beta(half, 0.5)))
+
+        def asymptotic(df):
+            # gammaln(x) ~ x * ln(x) - x - 0.5 * ln(x) + 0.5 * ln(2 * pi)
+            # psi(x) ~ ln(x) - 1 / (2 * x)
+            # B(x, y) ~ gamma(x) * y ** (-x)
+            # (x / 2) * (psi(x + 1) - psi(x)) ~ 0.5
+            h = 0.5 + 0.5 * sc.gammaln(0.5) + 0.5 * np.log(2)
+            return h
```

Comment:
An error crept in here: the factor `0.5` before `gammaln(0.5)` must be `1`. See the plot below.

Compared to wolfram alpha ([here](https://www.wolframalpha.com/input?i=asymptotic+expansion+%28d%2B1%29%2F2+*+%28digamma%28%28d%2B1%29%2F2%29+-+digamma%28d%2F2%29%29+%2B+log%28sqrt%28d%29+*+beta%28d%2F2%2C+1%2F2%29%29)), we get the same with `gammaln=log(pi)`. I think we should at least use the first term $1/df$ of the expansion provided by Wolfram Alpha. Maybe Wolfram Alpha can be a general alternative to the pen and paper calculations I have done in this area for these problems lately.

![image](https://user-images.githubusercontent.com/40656107/224531248-92ed1ea1-1f77-4c21-b1e7-3d151b728c53.png)

<details>

```
import numpy as np
from scipy import stats
from scipy import special as sc
import matplotlib.pyplot as plt

def asymptotic_wolfram_alpha(df):
    return 0.5 * (1 + np.log(2) + np.log(np.pi)) + 1/df

d = np.logspace(1, 20, 1000)
res_asymptotic = asymptotic(d)

plt.semilogx(d, stats.t.entropy(d), label="main")#, ls='dotted')
plt.semilogx(d, np.full_like(d, 0.5 * (1. + sc.gammaln(0.5) + np.log(2))), label="PR", ls='dashed')
plt.semilogx(d, np.full_like(d, 0.5 * (1. + 2 * sc.gammaln(0.5) + np.log(2))), label="PR corrected", ls='dashed')
plt.semilogx(d, asymptotic_wolfram_alpha(d), label="expansion wolfram alpha", ls='dashdot')
plt.ylim(0.5, 3)
plt.legend()
plt.show()
```

</details>

Reply:
>An error crept in here: the factor 0.5 before gammaln(0.5) must be 1. See the plot below.

Good catch. I meant to write `0.5 * np.log(np.pi)`, but I had a brain fart. I also added mpmath tests to verify that the formula was correct (which I should have done from the start, tbh).